core/sail/lmdb/src/main/java/org/eclipse/rdf4j/sail/lmdb/LmdbSailStore.java

```diff
@@ -891,5 +893,21 @@ public CloseableIteration<? extends Statement> getStatements(Resource subj, IRI
 				throw new SailException("Unable to get statements", e);
 			}
 		}
+
+		@Override
+		public CloseableIteration<? extends Statement> getStatements(StatementOrder statementOrder, Resource subj,
+				IRI pred, Value obj, Resource... contexts) throws SailException {
+			throw new UnsupportedOperationException("Not implemented yet");
+		}
+
+		@Override
+		public Set<StatementOrder> getSupportedOrders(Resource subj, IRI pred, Value obj, Resource... contexts) {
+			return Set.of();
+		}
+
+		@Override
+		public Comparator<Value> getComparator() {
+			throw new UnsupportedOperationException("Not implemented yet");
+		}
```

Comment:
@kenwenzel here are the three methods you need to implement to get merge join working.

Reply:
@hmottestad  I've added an initial implementation in:
https://github.com/kenwenzel/rdf4j/tree/GH-4819-merge-join

Unfortunately, `QueryBenchmark.complexQuery()` directly fails with an Exception. Maybe it helps to locate the problem.

Reply:
I fixed it, was an optimisation I had made where I hadn't quite accounted for all the edge cases.